src/main/scala/com/fulcrumgenomics/bam/api/SamRecord.scala

```diff
@@ -236,9 +236,83 @@ trait SamRecord {
   // transient attributes
   @inline final val transientAttrs: TransientAttrs = new TransientAttrs(this)
 
-  // TODO long-term: replace these two methods with methods on [[Cigar]] to save creating alignment blocks in memory
-  @inline final def refPosAtReadPos(pos: Int) = getReferencePositionAtReadPosition(pos)
-  @inline final def readPosAtRefPos(pos: Int, returnLastBaseIfDeleted: Boolean) = getReadPositionAtReferencePosition(pos, returnLastBaseIfDeleted)
+  /** Returns the 1-based reference position for the 1-based position in the read, or [[None]] if the reference does
+    * not map at the read position (e.g. insertion).
+    *
+    * @param pos the 1-based read position to query
+    * @param returnLastBaseIfInserted if the reference is an insertion, true to returned the previous reference base,
+    *                                 false to return None
+    * */
+  @inline final def refPosAtReadPos(pos: Int, returnLastBaseIfInserted: Boolean = false): Option[Int] = if (pos <= 0) None else {
+    var readPos      = 1
```

Comment:
I don't think this will handle leading soft-clipping correctly.  I believe it will return `rec.start` for any leading soft-clipped read position, and that seems wrong.

Reply:
fixed